open-sdg/open-sdg
# Site and indicator configuration forms

This PR adds configuration forms for site configuration as well as indicator configuration.

Testing this PR requires two things:
1. Use the latest jekyll-open-sdg-plugins code: `gem "jekyll-open-sdg-plugins", "1.2.0.pre.beta3"`
2. Add this to your site config:
```
create_config_forms:
    layout: config-builder
```

To get to the site configuration, look for the link in the footer menu.

To get to the indicator configuration, go to an indicator, click on the "Edit" tab, and then click on "Edit configuration".

The idea behind these configuration forms is that you'll make some changes, click the "Download configuration" button, and then upload the file to Github, replacing whatever file was already there. For site configuration, this would be a site_config.yml file in your data folder. For indicator, this would be somewhere in your data repository where indicator config could be stored.

With this approach, it is best to keep this automated configuration totally separate from other stuff. So, the site configuration should no longer be put into the _config.yml file. And indicator configuration should not be included with actual metadata.
